keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/PausedRepository.kt

```diff
@@ -17,24 +17,27 @@
  */
 package com.netflix.spinnaker.keel.persistence
 
+import com.netflix.spinnaker.keel.pause.Pause
+import com.netflix.spinnaker.keel.pause.PauseScope
+import java.time.Clock
+
 /**
  * A repository to track what scopes are paused, starting with application
  */
 interface PausedRepository {
+  val clock: Clock
+    get() = Clock.systemUTC()
+
+  fun getPause(scope: PauseScope, name: String): Pause?
 
-  fun pauseApplication(application: String)
+  fun pauseApplication(application: String, user: String)
   fun resumeApplication(application: String)
   fun applicationPaused(application: String): Boolean
 
-  fun pauseResource(id: String)
+  fun pauseResource(id: String, user: String)
   fun resumeResource(id: String)
```

Comment:
Should resume have the same capturing of user identity?

Reply:
Yes. I initially didn't think of adding it because there's no "resume" table to record it, but this will end up in the events. Was thinking it could be a separate PR since this one is pretty big already...

Reply:
I ended up adding this.